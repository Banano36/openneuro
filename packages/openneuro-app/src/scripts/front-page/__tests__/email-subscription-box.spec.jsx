import React from 'react'
import { mount } from 'enzyme'
import EmailSubscriptionBox, {
  modes,
  submitHandler,
} from '../email-subscription-box.jsx'

describe('EmailSubscriptionBox', () => {
  it('renders correctly in get-email mode', () => {
    const wrapper = mount(
      <EmailSubscriptionBox subscribe={() => {}} initialMode={modes.GET} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly in success mode', () => {
    const wrapper = mount(
      <EmailSubscriptionBox subscribe={() => {}} initialMode={modes.GET} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly in error mode', () => {
    const wrapper = mount(
      <EmailSubscriptionBox subscribe={() => {}} initialMode={modes.GET} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('EmailSubscriptionBox submitHandler', () => {
  const event = {
    preventDefault: () => {},
    target: { email: { value: 'email@email.test' } },
  }
  it('calls setMode with GET when in ERROR mode', () => {
    const subscribe = () => {}
    const currentMode = modes.ERROR
    const setMode = newMode => expect(newMode).toEqual(modes.GET)
    submitHandler(subscribe, currentMode, setMode)(event)
  })

  describe('properly handles setMode when email is submitted in GET mode', () => {
    const currentMode = modes.GET

    it('calls SUCCESS mode for successful submit', () => {
      const result = { data: { subscribeToNewsletter: true } }
      const subscribe = (someValidEmail, cb) => cb(result)
      const setMode = newMode => expect(newMode).toEqual(modes.SUCCESS)
      submitHandler(subscribe, currentMode, setMode)(event)
    })
    it('calls ERROR mode for unsuccessful submit', () => {
      const result = { data: { subscribeToNewsletter: false } }
      const subscribe = (someValidEmail, cb) => cb(result)
      const setMode = newMode => expect(newMode).toEqual(modes.ERROR)
      submitHandler(subscribe, currentMode, setMode)(event)
    })
  })
})
