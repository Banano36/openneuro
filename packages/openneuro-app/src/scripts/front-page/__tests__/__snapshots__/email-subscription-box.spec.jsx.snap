// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailSubscriptionBox renders correctly in error mode 1`] = `
<EmailSubscriptionBox
  initialMode={Symbol(get)}
  subscribe={[MockFunction]}
>
  <Styled(div)>
    <div
      className="css-e2r573"
    >
      <Styled(form)
        onSubmit={[Function]}
      >
        <form
          className="css-7r30sm"
          onSubmit={[Function]}
        >
          <Styled(h3)>
            <h3
              className="css-1dwdxm7"
            >
              Get Updates
            </h3>
          </Styled(h3)>
          <Styled(div)
            className="has-float-label"
          >
            <div
              className="has-float-label css-1wxcosw"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                placeholder=" "
                type="email"
              />
              <label
                htmlFor="email"
              >
                E-mail Address
              </label>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="css-1w6tkh5"
            >
              <button
                className="btn-blue"
                disabled={false}
                type="submit"
              >
                Subscribe
              </button>
            </div>
          </Styled(div)>
        </form>
      </Styled(form)>
    </div>
  </Styled(div)>
</EmailSubscriptionBox>
`;

exports[`EmailSubscriptionBox renders correctly in get-email mode 1`] = `
<EmailSubscriptionBox
  initialMode={Symbol(get)}
  subscribe={[MockFunction]}
>
  <Styled(div)>
    <div
      className="css-e2r573"
    >
      <Styled(form)
        onSubmit={[Function]}
      >
        <form
          className="css-7r30sm"
          onSubmit={[Function]}
        >
          <Styled(h3)>
            <h3
              className="css-1dwdxm7"
            >
              Get Updates
            </h3>
          </Styled(h3)>
          <Styled(div)
            className="has-float-label"
          >
            <div
              className="has-float-label css-1wxcosw"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                placeholder=" "
                type="email"
              />
              <label
                htmlFor="email"
              >
                E-mail Address
              </label>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="css-1w6tkh5"
            >
              <button
                className="btn-blue"
                disabled={false}
                type="submit"
              >
                Subscribe
              </button>
            </div>
          </Styled(div)>
        </form>
      </Styled(form)>
    </div>
  </Styled(div)>
</EmailSubscriptionBox>
`;

exports[`EmailSubscriptionBox renders correctly in success mode 1`] = `
<EmailSubscriptionBox
  initialMode={Symbol(get)}
  subscribe={[MockFunction]}
>
  <Styled(div)>
    <div
      className="css-e2r573"
    >
      <Styled(form)
        onSubmit={[Function]}
      >
        <form
          className="css-7r30sm"
          onSubmit={[Function]}
        >
          <Styled(h3)>
            <h3
              className="css-1dwdxm7"
            >
              Get Updates
            </h3>
          </Styled(h3)>
          <Styled(div)
            className="has-float-label"
          >
            <div
              className="has-float-label css-1wxcosw"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                placeholder=" "
                type="email"
              />
              <label
                htmlFor="email"
              >
                E-mail Address
              </label>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="css-1w6tkh5"
            >
              <button
                className="btn-blue"
                disabled={false}
                type="submit"
              >
                Subscribe
              </button>
            </div>
          </Styled(div)>
        </form>
      </Styled(form)>
    </div>
  </Styled(div)>
</EmailSubscriptionBox>
`;
